### ISSUE TITLE
Module $style doesn't contain mixins classes or "&name" classes

### ISSUE THREAD

**Issue by sheremet-va** [johnsoncodehk/volar#157]
For example if I have this code in my style tag (`scss, module`): 

```scss
.chevron {
   color: red;

   &Up {
      color: green;
   }
}
```

Type checking for `$style` fails because it doesn't contain the class `chevronUp`.

<img width="1109" alt="Снимок экрана 2021-04-25 в 13 14 49" src="https://user-images.githubusercontent.com/16173870/115989699-43889980-a5c8-11eb-9859-97e59d38c43b.png">

This also applies to global mixins and mixins in general. We have `@import "@/styles/_imports.scss"` in our `additionalData` for scss inside Vite config and import them like:

```scss
.input {
    @include input;
}
```

If mixing contains a class or classes, they don't appear inside `$style` variable. Right now I use `useCssStyle` in my `setup` method to bypass that, but that's not ideal.

**Comment by johnsoncodehk** [comment]
This is a design limitation. If we need to calculate all valid class names, this will be a compiler level job for each style language. (css, scss, less)

I suggest you use a single style block to shims the css module types.

```vue
<style module>
// you original styles
</style>

<style module>
.chevronUp { }
.chevronDown { }
// ...
</style>
```

**Comment by johnsoncodehk** [comment]
I found that `useCssModule()` type is `Record<string, string>`, maybe I should change `$style` types to `Record<string, string> & { chevron: string, ... }`, that allow all class name from `$style` in template, but we can't not get error report if we have some typing error.

**Comment by johnsoncodehk** [comment]
Fixed in 0.24.2.
